site/source/docs/optimizing/Optimizing-Code.rst

```diff
@@ -23,14 +23,15 @@ The optimization level you should use depends mostly on the current stage of dev
 
 - When first porting code, run *emcc* on your code using the default settings (without optimization). Check that your code works and :ref:`debug <Debugging>` and fix any issues before continuing.
 - Build with lower optimization levels during development for a shorter compile/test iteration cycle (``-O0`` or ``-O1``).
-- Build with ``-O2`` or ``-O3`` when releasing your code.  ``-O3`` builds are even more optimized than ``-O2``, but at the cost of significantly longer compilation time.
+- Build with ``-O2`` to get most optimizations.
+- Build with ``-O3`` or ``-Os`` when releasing your code. ``-O3`` builds are even more optimized than ``-O2``, but at the cost of significantly longer compilation time. ``-Os`` builds focus more on size than speed (and also can increase compilation times like ``-O3``).
```

Comment:
It's probably also worth noting that O3 can make tradeoffs in favor of speed (i.e. speed of generated code) at the cost of larger code size. For, e.g. server applications this might be what you want but for web deployment it's less clear. I might recommend that users consider O2 as the default and not pick O3 unless they measure the costs and benefits for their particular apps.

Reply:
Makes sense, adding.